pkg/kubefed2/util/util.go

```diff
@@ -113,9 +113,9 @@ func ClusterServiceAccountName(joiningClusterName, hostClusterName string) strin
 	return fmt.Sprintf("%s-%s", joiningClusterName, hostClusterName)
 }
 
-// ClusterRoleName returns the name of a ClusterRole and its associated
-// ClusterRoleBinding that are used to allow the service account to
-// access necessary resources on the cluster.
-func ClusterRoleName(serviceAccountName string) string {
+// RoleName returns the name of a Role or ClusterRole and its
+// associated RoleBinding or ClusterRoleBinding that are used to allow
+// the service account to access necessary resources on the cluster.
+func RoleName(serviceAccountName string) string {
```

Comment:
Shall we enhance this a bit by adding namespace to this as well?

Perhaps adding two util functions maybe better? One is for `RoleName` with `namespace` and another is `ClusterRoleName` without `namespace`?

```
func RoleName(serviceAccountName, namespace string) string {
}
```
```
func ClusterRoleName(serviceAccountName string) string {
}
```

Reply:
I call YAGNI on both counts.  Having the same name for different resources is fine, they are entirely different contexts.  If/when we see a use case for clusters participating in multiple federations, I can imagine a single cluster role allowing access to `healthz` being defined once for the cluster by an administrator rather than relying on a tool like kubefed.